typespec-extension/src/code-model-builder.ts

```diff
@@ -1730,37 +1733,37 @@ export class CodeModelBuilder {
     type.members.forEach((it) => choices.push(new ChoiceValue(it.name, this.getDoc(it), it.value ?? it.name)));
 
     if (sealed) {
-      return this.codeModel.schemas.add(
-        new SealedChoiceSchema(name, this.getDoc(type), {
-          summary: this.getSummary(type),
-          choiceType: valueType as any,
-          choices: choices,
-          language: {
-            default: {
-              namespace: namespace,
-            },
-            java: {
-              namespace: getJavaNamespace(namespace),
-            },
+      const sealedChoiceSchema = new SealedChoiceSchema(name, this.getDoc(type), {
+        summary: this.getSummary(type),
+        choiceType: valueType as any,
+        choices: choices,
+        language: {
+          default: {
+            namespace: namespace,
           },
-        }),
-      );
+          java: {
+            namespace: getJavaNamespace(namespace),
+          },
+        },
+      });
+      sealedChoiceSchema.crossLanguageDefinitionId = getCrossLanguageDefinitionId(type);
```

Comment:
Totally perisomal opinion and likely very subjective: I kind of inclined to put `crossLanguageDefinitionId` into `language.default.crossLanguageDefinitionId` so it applies to any code model object (`crossLanguageDefinitionId` being a pretty universal identity, like `namespace`+`name` and `description` etc.).

No need to change, we can discuss and adjust in follow-up PR.